### PR TITLE
update only if i'm the touched item

### PR DESCRIPTION
**Pull request by DavidBabel**
see https://github.com/jaredpalmer/formik/issues/1103

### CODE REVIEW

File: src/FastField.tsx [hunk 1]
```diff
@@ -110,20 +110,22 @@ class FastFieldInner<Values = {}, Props = {}> extends React.Component<
...
-      getIn(this.props.formik.touched, this.props.name) !==
-        getIn(props.formik.touched, this.props.name) ||
-      Object.keys(this.props).length !== Object.keys(props).length ||
-      this.props.formik.isSubmitting !== props.formik.isSubmitting
-    ) {
-      return true;
-    } else {
-      return false;
     }
+    if (props.formik.lastTouched === this.props.name) {
```
Comment (brodybits) [inline src/FastField.tsx]: Why not put this all in a single return statement, something like this?

```js
    return (props.formik.lastTouched === this.props.name) && (
        this.props.formik.isSubmitting !== props.formik.isSubmitting ||
        Object.keys(this.props).length !== Object.keys(props).length ||
        getIn(this.props.formik.values, this.props.name) !==
          getIn(props.formik.values, this.props.name) ||
        getIn(this.props.formik.errors, this.props.name) !==
          getIn(props.formik.errors, this.props.name) ||
        getIn(this.props.formik.touched, this.props.name) !==
          getIn(props.formik.touched, this.props.name)
      );
```

Comment (brodybits) [reply src/FastField.tsx]: Or more drastically rewrite the function body like this:

```js
    return this.props.shouldUpdate) ?
      this.props.shouldUpdate(props, this.props) :
      (props.formik.lastTouched === this.props.name) && (
        this.props.formik.isSubmitting !== props.formik.isSubmitting ||
        Object.keys(this.props).length !== Object.keys(props).length ||
        getIn(this.props.formik.values, this.props.name) !==
          getIn(props.formik.values, this.props.name) ||
        getIn(this.props.formik.errors, this.props.name) !==
          getIn(props.formik.errors, this.props.name) ||
        getIn(this.props.formik.touched, this.props.name) !==
          getIn(props.formik.touched, this.props.name)
      );
```

Comment (DavidBabel) [reply src/FastField.tsx]: i hardly prefer your first suggestion for reading purpose.

note: is it a fast review because it relies on fast fields ? :p

Comment (DavidBabel) [reply src/FastField.tsx]: changed it

**Comment by tlenclos** [comment]
We have a big form in our app and this PR is improving performance 👍 
The user experience is still very bad for our user but this is a move in the right direction.

**Comment by jaredpalmer** [comment]
While this would work for Formik v1's registration, i think it has some problems with registration in v2. In v1 we naively assume there is 1:1 relation ship with fields to components. This is why we didn't implement ErrorMessage with field (because that would break). However, in v2, this won't be the case. This trick could still work though, the value would just include some kind of random number or uuid.

**Comment by jaredpalmer** [comment]
However, we may not even really need registration (or rather it may become hard to understand) with fiber/suspense and with upcoming `useField`. Not sure yet.

**Comment by jaredpalmer** [comment]
Closing in favor of #2749

**Pull request closed without merging**